### PR TITLE
Make X509Certificate.Verify test OuterLoop and add diagnostics

### PR DESCRIPTION
**Pull request by steveharter**
Addresses https://github.com/dotnet/corefx/issues/13979 CertTests.TestVerify test failed in CI

Perhaps internet\network was down, time was wrong on local machine, or some other issue so moving test to outerloop to minimize issues with (rare) random failures.

Uses ITestOutputHelper to capture information in .xml output (is there a better\different prescriptive way, like standard debug messages?)

cc @bartonjs

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs [hunk 1]
```diff
@@ -80,29 +88,65 @@ public static void X509Cert2Test()
         }
 
 #if netstandard17
-        [Fact]
-        public static void TestVerify()
+        [OuterLoop( /* May require using the network, to download CRLs and intermediates */)]
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs]: I'm pretty sure that you still need [Fact] with [OuterLoop].

And the new style (which didn't exist when I was doing the C-style comment in parens) is to use a literal string there.

File: src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs [hunk 2]
```diff
@@ -4,13 +4,21 @@
...
+        private readonly ITestOutputHelper _log;
+
+        public CertTests(ITestOutputHelper output)
+        {
+            _log = output;
+        }
+
         [Fact]
-        public static void X509CertTest()
+        public void X509CertTest()
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs]: I'm not sure about this, but I feel like it'll still run static methods when there's a mix of static and instance.  So you can keep the test class object instantiations down by leaving the rest static (xunit creates a new instance per instance-fact-method... possibly a new instance per instance-theory-case).

File: src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs (hunk 1 above)
Comment (steveharter) [inline src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs]: Oops on no [Fact]. Added the OuterLoop string description

**Comment by steveharter** [comment]
@bartonjs any additional feedback?

File: src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs [hunk 3]
```diff
@@ -81,22 +89,59 @@ public static void X509Cert2Test()
...
+                using (var chain = new X509Chain())
+                {
+                    if (!chain.Build(cert))
+                    {
+                        foreach (X509ChainStatus chainStatus in chain.ChainStatus)
+                        {
+                            _log.WriteLine(string.Format($"X509Certificate2.Verify error: {testName}, {chainStatus.Status}, {chainStatus.StatusInformation}"));
+                        }
+                    }
+                }
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs]: You may want to log if Build returns true (which could happen if a networking error didn't happen this time)

**Review by bartonjs — APPROVED** [review]

**Pull request merged**